core/src/com/unciv/ui/pickerscreens/UnitRenamePopup.kt

```diff
@@ -0,0 +1,49 @@
+package com.unciv.ui.pickerscreens
+
+import com.badlogic.gdx.graphics.Color
+import com.badlogic.gdx.scenes.scene2d.ui.Table
+import com.badlogic.gdx.utils.Align
+import com.unciv.UncivGame
+import com.unciv.logic.map.MapUnit
+import com.unciv.models.TutorialTrigger
+import com.unciv.models.UncivSound
+import com.unciv.models.ruleset.unique.UniqueType
+import com.unciv.models.ruleset.unit.Promotion
+import com.unciv.models.translations.tr
+import com.unciv.ui.images.ImageGetter
+import com.unciv.ui.popup.AskTextPopup
+import com.unciv.ui.utils.BaseScreen
+import com.unciv.ui.utils.KeyCharAndCode
+import com.unciv.ui.utils.RecreateOnResize
+import com.unciv.ui.utils.extensions.isEnabled
+import com.unciv.ui.utils.extensions.onClick
+import com.unciv.ui.utils.extensions.surroundWithCircle
+import com.unciv.ui.utils.extensions.toLabel
+import com.unciv.ui.utils.extensions.toTextButton
+
+class UnitRenamePopup(val screen: BaseScreen, val unit: MapUnit, val actionOnClose: ()->Unit) {
+    init {
+        AskTextPopup(
+            screen,
+            label = "Choose name for [${unit.baseUnit.name}]",
+            icon = ImageGetter.getUnitIcon(unit.name).surroundWithCircle(80f),
+            defaultText = if (unit.instanceName == null) {
+                unit.baseUnit.name
+            } else {
+                unit.instanceName.toString()
+            },
```

Comment:
These lines can be replaced with just one:
```Kotlin
defaultText = unit.instanceName ?: unit.baseUnit.name.tr(),
```
Do not forget that the initial suggestion for the name should be shown in the language the user sees its type, so `.tr()` is needed there.